contracts/lbp/LBPWrapper.sol

```diff
@@ -33,104 +33,112 @@ contract LBPWrapper {
      * @param _endTime             end time
      * @param _endWeights          array of end weights for respective tokens
-     * @param _admin               address of admin/owner of LBP
-     * @param _userData            extra data required by LBP
      */
-    function deployLbpFromFactory(
+    function initializeLBP(
+        address _LBPFactory,
         string memory _name,
         string memory _symbol,
         IERC20[] memory _tokens,
-        uint256[] memory _amounts,
         uint256[] memory _weights,
-        bool _swapEnabledOnStart,
         uint256 _startTime,
         uint256 _endTime,
         uint256[] memory _endWeights,
-        address _admin,
-        bytes memory _userData
-    ) public onlyOwner returns (address) {
-        // to handle stack overflow
-        {
-            address vault = address(LBPFactory.getVault());
-            for (uint8 i; i < _tokens.length; i++) {
-                IERC20(_tokens[i]).approve(vault, _amounts[i]);
-            }
-        }
+        uint256 _swapFeePercentage
+    ) external returns (address) {
+        require(!initialized, "LBPWrapper: already initialized");
+
+        initialized = true;
+        owner = msg.sender;
 
-        lbp = LBPFactory.create(
-            _name,
-            _symbol,
-            _tokens,
-            _weights,
-            swapFeePercentage,
-            address(this),
-            _swapEnabledOnStart
+        lbp = ILBP(
+            ILBPFactory(_LBPFactory).create(
+                _name,
+                _symbol,
+                _tokens,
+                _weights,
+                _swapFeePercentage,
+                address(this),
+                false // swapEnabledOnStart is set to false at pool creation
+            )
         );
 
-        // to handle stack overflow
-        {
-            ILBP(lbp).updateWeightsGradually(_startTime, _endTime, _endWeights);
+        lbp.updateWeightsGradually(_startTime, _endTime, _endWeights);
+
+        return address(lbp);
+    }
+
+    /**
+     * @dev                          approve tokens for the vault and join pool to provide liquidity
+     * @param _tokens                array of tokens sorted for the LBP
+     * @param _amounts               amount of tokens to add to the pool to provide liquidity
+     * @param _receiver              address who will receive the Balancer Pool Tokens (BPT)
+     * @param _fromInternalBalance   fund tokens from the internal user balance
+     * @param _userData              userData specifies the type of join
+     */
+    function fundPool(
+        IERC20[] memory _tokens,
+        uint256[] memory _amounts,
+        address _receiver,
+        bool _fromInternalBalance,
+        bytes memory _userData
+    ) public onlyOwner {
+        require(!poolFunded, "LBPWrapper: pool has already been funded");
+
+        IVault vault = lbp.getVault();
+        for (uint8 i; i < _tokens.length; i++) {
+            _tokens[i].approve(address(vault), _amounts[i]);
```

Comment:
What if the admin wants to fund using internal balance, we should add an if condition, to check for possible revert.